lib/model/QubitRepository.php

```diff
@@ -124,43 +124,54 @@ public function save($connection = null)
     return $this;
   }
 
+  public function getRelatedInformationObjectIds()
+  {
+    $sql = "SELECT id FROM ". QubitInformationObject::TABLE_NAME ." WHERE repository_id=:repository_id";
+
+    $params = array(':repository_id' => $this->id);
+    return QubitPdo::fetchAll($sql, $params, array('fetchMode' => PDO::FETCH_COLUMN));
+  }
+
   public function updateRelatedIos()
   {
-    $ioIds = array();
     $context = sfContext::getInstance();
     $env = $context->getConfiguration()->getEnvironment();
```

Comment:
Real nitpick: the context and env could be obtained after checking the IO ids, to avoid it if there are no related IOs.

Reply:
True. Moved those and got rid of $env variable as env is only referred to once.